LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/NetworkServerConfiguration.cs

```diff
@@ -154,14 +154,28 @@ public static NetworkServerConfiguration CreateFromEnvironmentVariables()
             var envVars = new CaseInsensitiveEnvironmentVariables(Environment.GetEnvironmentVariables());
             config.ProcessingDelayInMilliseconds = envVars.GetEnvVar("PROCESSING_DELAY_IN_MS", config.ProcessingDelayInMilliseconds);
             config.RunningAsIoTEdgeModule = !envVars.GetEnvVar("CLOUD_DEPLOYMENT", false);
-            config.IoTHubHostName = envVars.GetEnvVar("IOTEDGE_IOTHUBHOSTNAME", string.Empty);
+
+            var iotHubHostName = envVars.GetEnvVar("IOTEDGE_IOTHUBHOSTNAME", string.Empty);
+            if (string.IsNullOrEmpty(iotHubHostName))
+            {
+                iotHubHostName = envVars.GetEnvVar("IOTHUBHOSTNAME", string.Empty);
+            }
+            config.IoTHubHostName = !string.IsNullOrEmpty(iotHubHostName) ? iotHubHostName : throw new InvalidOperationException("Either 'IOTEDGE_IOTHUBHOSTNAME' or 'IOTHUBHOSTNAME' environment variable should be populated");
+
             config.GatewayHostName = envVars.GetEnvVar("IOTEDGE_GATEWAYHOSTNAME", string.Empty);
             config.EnableGateway = envVars.GetEnvVar("ENABLE_GATEWAY", true);
             if (!config.RunningAsIoTEdgeModule && config.EnableGateway)
             {
                 throw new NotSupportedException("ENABLE_GATEWAY cannot be true if RunningAsIoTEdgeModule is false.");
             }
-            config.GatewayID = envVars.GetEnvVar("IOTEDGE_DEVICEID", string.Empty);
+
+            var gatewayId = envVars.GetEnvVar("IOTEDGE_DEVICEID", string.Empty);
+            if (string.IsNullOrEmpty(gatewayId))
+            {
+                gatewayId = envVars.GetEnvVar("HOSTNAME", string.Empty);
+            }
+            config.GatewayID = !string.IsNullOrEmpty(gatewayId) ? gatewayId : throw new InvalidOperationException("Either 'IOTEDGE_DEVICEID' or 'HOSTNAME' environment variable should be populated");
```

Comment:
[nit] consider extracting the duplicated structure (fallback env variables) into a local function